awscli/shorthand.py

```diff
@@ -332,17 +361,51 @@ def _consume_quoted(self, regex, escaped_char=None):
             value = value.replace("\\\\", "\\")
         return value
 
-    def _double_quoted_value(self):
-        return self._consume_quoted(self._DOUBLE_QUOTED, escaped_char='"')
+    def _double_quoted_value(self, parent_key=None, resolve_paramfiles=False):
+        return self._resolve_paramfile(
+            self._consume_quoted(self._DOUBLE_QUOTED, escaped_char='"'),
+            resolve_paramfiles,
+            parent_key
+        )
 
-    def _second_value(self):
+    def _second_value(self, parent_key, resolve_paramfiles=False):
         if self._current() == "'":
-            return self._single_quoted_value()
+            return self._single_quoted_value(parent_key, resolve_paramfiles)
         elif self._current() == '"':
-            return self._double_quoted_value()
+            return self._double_quoted_value(parent_key, resolve_paramfiles)
         else:
             consumed = self._must_consume_regex(self._SECOND_VALUE)
-            return consumed.replace('\\,', ',').rstrip()
+            return self._resolve_paramfile(
+                consumed.replace('\\,', ',').rstrip(),
+                resolve_paramfiles,
+                parent_key
+            )
+
+    def _resolve_paramfile(self, val, resolve_paramfiles, parent_key=None):
+        # If resolve_param_files is True, this function tries to resolve val to a
+        # paramfile (i.e. a file path prefixed with a key of LOCAL_PREFIX_MAP).
+        # If val is a paramfile, returns the contents of the file (retrieved
+        # according to the spec of get_paramfile).
+        # If val is not a paramfile, returns val.
+        # If resolve_param_files is False, returns val.
+        # if a parent_key is supplied, then a warning will be printed if resolve_param_files=False
+        # and the value is parsed to a string that starts with a file prefix
+        if (resolve_paramfiles and
+                (paramfile := get_paramfile(val, LOCAL_PREFIX_MAP)) is not None):
+            return paramfile
+        if parent_key is not None and not resolve_paramfiles:
+            self._print_file_warnings_if_prefixed(parent_key, val)
+        return val
+
+    def _print_file_warnings_if_prefixed(self, key, val):
+        for prefix in LOCAL_PREFIX_MAP.keys():
+            if val.startswith(prefix):
+                _LOGGER.warning(f'Usage of the {prefix} prefix was detected '
```

Comment:
For reviewer, initially I planned to use `uni_print` instead of `logging` for visibility to the user: I wanted it to print to the command line even without the `--debug` flag. However, my concern is that there is no theoretical upper bound on the number of times this gets printed for a single command. It would print once per `file://` usage without the `@=` operator. This could spam the command line if a list of strings prefixed with `file://` is supplied. 

For this reason, I chose to log the warning only in debug mode. Any feedback on this?